neuraxle/metaopt/auto_ml.py

```diff
@@ -714,7 +714,8 @@ def __init__(
             callbacks: List[BaseCallback] = None,
             refit_scoring_function: Callable = None,
             print_func: Callable = None,
-            cache_folder_when_no_handle=None
+            cache_folder_when_no_handle=None,
+            raise_all_error_types=False
```

Comment:
typing: what type should be this variable?
Would be good to type the two variables here despite only 1 is new. 

I also suggest the following renaming:
```suggestion
            do_catch_all_error_types=False
```

Reply:
I disagree with the name change. We always want to raise the following errors : (SystemError, SystemExit, EOFError, KeyboardInterrupt). Thus the name "do_catch_all_error_types" would be misleading as we never catch those. 

I can rename it to "do_raise_all_error_types" if you'd like.

Reply:
What I find confusing is the fact that it talks about raising whereas in this part of code all we do is managing what we catch. This way, if `do_catch_all_error_types=False`, it is an "all" so it doesn't exclude the "any" case. My suggestion still seems better in my opinion

Reply:
Renamed it to continue_loop_on_error which indicates what it does in term of effect in the loop execution. I'll also update the docstring.